src/migrate/expressions.ts

```diff
@@ -18,19 +18,41 @@ export function expressions(style: StyleSpecification) {
     eachLayer(style, (layer: LayerSpecification & { filter?: FilterSpecification }) => {
         if (layer.filter) {
             layer.filter = convertFilter(layer.filter);
+            layer.filter = convertGeometryType(layer.filter);
         }
     });
 
     eachProperty(style, {paint: true, layout: true}, ({path, value, reference, set}) => {
-        if (isExpression(value)) return;
-        if (typeof value === 'object' && !Array.isArray(value)) {
-            set(convertFunction(value, reference));
+        if (isExpression(value)) {
+            set(convertGeometryType(value));
+        } else if (typeof value === 'object' && !Array.isArray(value)) {
+            set(convertGeometryType(convertFunction(value, reference)));
             converted.push(path.join('.'));
         } else if ((reference as any).tokens && typeof value === 'string') {
-            set(convertTokenString(value));
+            set(convertGeometryType(convertTokenString(value)));
         }
     });
 
     return style;
 }
 
+const v21NonMultiGeometryType = [
```

Comment:
Is there a way to test the performance of slice? As I understand this is run on every single object, isn't it?